File: src/main/java/work/lclpnet/ap2/impl/game/data/entry/SimpleOrderDataEntry.java
package work.lclpnet.ap2.impl.game.data.entry;

import org.jetbrains.annotations.Nullable;
import work.lclpnet.ap2.api.game.data.DataEntry;
import work.lclpnet.ap2.api.game.data.SubjectRef;
import work.lclpnet.kibu.translate.TranslationService;
import work.lclpnet.kibu.translate.text.TranslatedText;

public record SimpleOrderDataEntry<Ref extends SubjectRef>(Ref subject, int order) implements DataEntry<Ref> {

    @Nullable
    @Override
    public TranslatedText toText(TranslationService translationService) {
        return null;
    }

    @Override
    public boolean scoreEquals(DataEntry<Ref> _other) {
        if (_other instanceof SimpleOrderDataEntry<Ref> other) {
            return order == other.order;
        }

        return false;
    }
}

File: src/lib/src/main/kotlin/work/lclpnet/ap2/game/data/entry/SimpleOrderDataEntry.kt
package work.lclpnet.ap2.game.data.entry

import work.lclpnet.ap2.api.game.data.DataEntry
import work.lclpnet.ap2.api.game.data.SubjectRef
import work.lclpnet.kibu.translate.Translations
import work.lclpnet.kibu.translate.text.TranslatedText

@JvmRecord
data class SimpleOrderDataEntry<Ref : SubjectRef>(
    override val subject: Ref,
    val order: Int,
    val data: TranslatedText?
) : DataEntry<Ref> {

    constructor(subject: Ref, order: Int) : this(subject, order, null)

    override fun toText(translationService: Translations): TranslatedText? {
        return data
    }

    override fun scoreEquals(other: DataEntry<Ref>): Boolean {
        if (other is SimpleOrderDataEntry<Ref>) {
            return order == other.order
        }

        return false
    }
}
